akka-remote/src/main/scala/akka/remote/artery/compress/OutboundActorRefCompression.scala

```diff
@@ -101,6 +107,39 @@ private[remote] class OutboundCompressionTable[T](system: ActorSystem, remoteAdd
 
 }
 
+class HashScanIndexMap[T](val size: Int) {
```

Comment:
Yes, it's a linear scan through the int-array + equals check if hashcode is correct.
Currently we estimate the table to be rather small, though I'm not sure keeping it very small (~256) is the best we could do.
For larger ones I expect the FastUtil open hashing map to be better.
Didn't benchmark for higher numbers yet.

In this PR we have all impls ready, so we can compare :)